subprojects/java21/src/test/java/org/mockito/internal/stubbing/answers/DeepStubReturnsEnumJava21Test.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2023 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockito.internal.stubbing.answers;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.*;
+
+import org.junit.Test;
+import org.mockito.exceptions.base.MockitoException;
+
+public class DeepStubReturnsEnumJava21Test {
+
+    @Test
+    public void cant_mock_enum_class_in_Java21_Issue_2984() {
+        assertThatThrownBy(
+                        () -> {
+                            mock(TestEnum.class);
+                        })
+                .isInstanceOf(MockitoException.class)
+                .hasMessage(
+                        "\nMockito cannot mock this class: class org.mockito.internal.stubbing.answers.DeepStubReturnsEnumJava21Test$TestEnum.\n"
+                                + "Sealed abstract enums can't be mocked. Since Java 15 abstract enums are declared sealed, which prevents mocking.\n"
+                                + "You can still return an existing enum literal from a stubbed method call.")
+                .hasCauseInstanceOf(MockitoException.class);
```

Comment:
```suggestion
                .hasMessageContaining("Sealed abstract enums can't be mocked.")
```

This is sufficient to test it throws the correct. It allows us to more easily tweak the error message without having to constantly update the tests as well.

Reply:
Same with the other exception assertions